### DISCUSSION TITLE
About the support of 6-axis force sensor/ contact sensor for plane contact

### DISCUSSION THREAD

**Discussion by annan-tang** [isaac-sim/IsaacLab#139] [Q&A]
Hi, Thank you very much for the great work! I am a researcher on humanoid robots. Previously, I used Isaacgym as the learning environment. For the humanoid robot, especially one that has a plane foot, the 6-axis contact force is essential for humanoid research. However, the force sensor in Isaac gym is unreliable, and I think there are no plans to fix such bug in Isaac gym. Thanks to the new feature of Isaac Sim and orbit, I want to transfer my project to orbit. Is the force sensor in orbit(or Isaac sim) reliable?

**Comment by Milad-Rakhsha** [comment]
You can use a fixed joint to model an FT sensor. You can read the joint forces through Isaac core API in the latest release 2023.1
See also this documentation page
https://docs.omniverse.nvidia.com/isaacsim/latest/features/sensors_simulation/isaac_sim_sensors_physics_articulation_force.html

**Comment by annan-tang** [comment]
Thank you very much. I will test the force_sensor function for plane contact(not point contact) on the isaac sim 2023.01. Actually, in Isaac gym, there exists similar force_sensor functions for 6-axis force measurements. Such as the usage in legged_gym:[[README.md](https://github.com/leggedrobotics/legged_gym/tree/master#readme)->known issue](https://github.com/leggedrobotics/legged_gym/tree/master#:~:text=The%20contact%20forces,with%20sensor_options.enable_forward_dynamics_forces). Although it works well for point contact(for quadrupedal legs), it gives wrong result for humanoid foot(I attached it on the foot surface).  From the paper of Orbit, I find that both isaac gym and isaac sim use the same version of PhysX(version 5). I think it is a bug(or limitation) of PhysX. So I open this discussion for this question: is the isaac gmy(preview4) use totally same PhysX as isaac sim. I'm just wondering whether such bugs/limitations/functionality are updated for physX backend inside the new isaac sim release 2023.1 or current Orbit framework.

**Comment by Mayankm96** [comment]
Hi, 

Thanks for bringing this up. Isaac Gym uses Physx 5.0 or 5.1 (from what I know), while Isaac Sim 2023.1 uses PhysX 5.3. A lot of changes and improvements have been made in PhysX since, so they are not exactly the same engine. You can view more information on the PhysX side of things [here](https://nvidia-omniverse.github.io/PhysX/physx/5.3.0/docs/MigrationTo53.html).

Coming back to your original question: This is something we would need to ask the PhysX team to understand how the engine is handling planar contacts and what exactly it reports in such situations for the net contact force. I will update the thread here once I receive some more information.
